### PR TITLE
Exposed parameter 'UnregisterUnusedVersions'

### PR DESCRIPTION
**Pull request by sidharthmittal**
Upgrade Parameter UnregisterUnusedVersions exposed in this commit.

### CODE REVIEW

File: Tasks/ServiceFabricDeploy/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -25,6 +25,8 @@
   "loc.input.help.skipUpgradeSameTypeAndVersion": "Indicates whether an upgrade will be skipped if the same application type and version already exists in the cluster, otherwise the upgrade fails during validation. If enabled, re-deployments are idempotent.",
   "loc.input.label.skipPackageValidation": "Skip package validation",
   "loc.input.help.skipPackageValidation": "Indicates whether the package should be validated or not before deployment.",
+  "loc.input.label.unregisterUnusedVersions": "Unregister Unused Versions",
+  "loc.input.help.unregisterUnusedVersions": "This will remove all unused type versions after deployment.",
```
Comment (MikkelHegn) [inline Tasks/ServiceFabricDeploy/Strings/resources.resjson/en-US/resources.resjson]: I would rephrase to "Indicates whether all unused versions of the application type will be removed after an upgrade."

Comment (MikkelHegn) [inline Tasks/ServiceFabricDeploy/Strings/resources.resjson/en-US/resources.resjson]: It actually looks like we do this no matter if the upgrade succeeded or not.

Comment (sidharthmittal) [inline Tasks/ServiceFabricDeploy/Strings/resources.resjson/en-US/resources.resjson]: Please go ahead and make the required change.

Comment (sidharthmittal) [inline Tasks/ServiceFabricDeploy/Strings/resources.resjson/en-US/resources.resjson]: Do you want me to change it and update the pull request?

File: Tasks/ServiceFabricDeploy/deploy.ps1 [hunk 2]
```diff
@@ -31,6 +31,7 @@ try {
     $compressPackage = [System.Boolean]::Parse((Get-VstsInput -Name compressPackage))
     $skipUpgrade =  [System.Boolean]::Parse((Get-VstsInput -Name skipUpgradeSameTypeAndVersion))
     $skipValidation =  [System.Boolean]::Parse((Get-VstsInput -Name skipPackageValidation))
+    $unregisterUnusedVersions = [System.Boolean]::Parse((Get-VstsInput -Name unregisterUnusedVersions))
```
Comment (NCarlsonMSFT) [inline Tasks/ServiceFabricDeploy/deploy.ps1]: Remember to run `node make.js test --task ServiceFabricDeploy` to test the change. You'll need to add mocking of the new input to the existing tests.

File: Tasks/ServiceFabricDeploy/task.loc.json [hunk 3]
```diff
@@ -128,6 +128,15 @@
       "helpMarkDown": "ms-resource:loc.input.help.skipPackageValidation"
     },
     {
+      "name": "unregisterUnusedVersions",
+      "type": "boolean",
+      "label": "ms-resource:loc.input.label.unregisterUnusedVersions",
+      "defaultValue": "false",
```
Comment (NCarlsonMSFT) [inline Tasks/ServiceFabricDeploy/task.loc.json]: To maintain compatibility for existing users this should be true.

File: Tasks/ServiceFabricDeploy/deploy.ps1 (hunk 2 above)
Comment (sidharthmittal) [inline Tasks/ServiceFabricDeploy/deploy.ps1]: Not able to run test.

image

Comment (NCarlsonMSFT) [inline Tasks/ServiceFabricDeploy/deploy.ps1]: Did you run all the set-up steps in [contribute](https://github.com/Microsoft/vsts-tasks/blob/master/docs/contribute.md), in particular: [make-dependencies](https://github.com/Microsoft/vsts-tasks/blob/master/docs/contribute.md#make-dependencies)?

Comment (sidharthmittal) [inline Tasks/ServiceFabricDeploy/deploy.ps1]: Missed that but, still facing this issue.

image

Comment (NCarlsonMSFT) [inline Tasks/ServiceFabricDeploy/deploy.ps1]: Looking at the output it looks like you using the Current version of node. Could you try using the LTS version instead? That is working on my machines.

Comment (sidharthmittal) [inline Tasks/ServiceFabricDeploy/deploy.ps1]: Version was the issues. Done the requested changes. Please check!

**Comment by joechiu-msft** [comment]
This is exactly what we need, so we can retain older versions for potential rollback. Good job, @sidharthmittal

**Comment by NCarlsonMSFT** [comment]
@MikkelHegn, @bishal-pdMSFT any objections to merging?

**Comment by bishal-pdMSFT** [comment]
@NCarlsonMSFT In my opinion, this parameter should not be first in "upgrade" group.

**Comment by sidharthmittal** [comment]
@bishal-pdMSFT  Have made the change you suggested. Moved it to an apt position.
Please merge as soon as possible. We are waiting for this property to be exposed
